### PR TITLE
Game controller

### PR DESCRIPTION
**Pull request by Dylanflini**
# Summary

## References

# Self review

- [x] I have reviewed my PR changes and left comments/questions when necessary
- [ ] I have added a screenshot for the applied changes.
- [ ] I have followed the style guidelines of this project.
- [x] I have added and ran the tests locally.
- [ ] I have ran the test coverage reports locally.
- [ ] I have made corresponding changes to the documentation (README and API).
- [ ] I have added the abstraction layers needed to separate responsibilities.

## Impact of these changes

This changes through `Component` impacts:

- `Component1`
- `Component2`
- `Component3`

## Evidences of expected changes

Add evidences here about each impact generated by your changes

# Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hot fix (fixes an issue in master branch)
- [ ] Docs (Documentation only changes)
- [ ] Style (Changes that do not affect the meaning of the code, formatting, missing semi-colons, etc)
- [ ] Refactor (Changes that neither fixes a bug or adds a feature)
- [ ] Performance (Changes that improve performance)
- [ ] Chore (Changes to the build process or auxiliary tools and libraries)

# Squash and merge

We are following [Angular commit messages guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines)

```
<type>(<scope>): <Jira-Issue subject>
<BLANK LINE>
<optional body>
<BLANK LINE>
<optional footer>
```

Types

```
feat: A new feature
fix: A bug fix
docs: Documentation only changes
style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
refactor: A code change that neither fixes a bug nor adds a feature
perf: A code change that improves performance
test: Adding missing or correcting existing tests
chore: Changes to the build process or auxiliary tools and libraries such as documentation generation
```

Make sure you squash and merge your feature/bugfix branches into develop/release/hotfix branches using the suggested format.

Some examples.

```
feat: DP-1234 Edit assessment information
Frontend changes for editing an assessment:
- improve assign modal to receive an existing assessment
- integrate the reasign end point.
- couple of warnings removed
BREAKING CHANGE: the assign modal now receives 3 required parameters
```

More examples [here](https://www.conventionalcommits.org/en/v1.0.0-beta.2/#summary)

### CODE REVIEW

File: src/minesweeper/use-cases/startGame/startGame.validations.ts [hunk 1]
```diff
@@ -1,23 +1,30 @@
...
 
-export enum ErrorStartGame {
+export enum StartGameErrorMessages {
   BOMBS_GREATER_THAN_ZERO = 'bombs must be greater than 0',
   ROWS_GREATER_THAN_ZERO = 'rows must be greater than 0',
   COLUMNS_GREATER_THAN_ZERO = 'columns must be greater than 0',
-  ROWS_AND_COLUMNS_GREATER_THAN_ONE = 'rows and columns must be greater than 1',
-  BOMBS_GREATER_THAN_TOTAL_CELLS = 'bombs must be less than total cells',
+  BOMBS_LESS_THAN_TOTAL_CELLS = 'bombs must be less than total cells',
+  BOMBS_IS_INTEGER = 'bombs must be a integer number',
```
Comment (danielglobant) [inline src/minesweeper/use-cases/startGame/startGame.validations.ts:8]: Is not integer, is a natural number. `Integer` includes negative numbers

File: src/minesweeper/use-cases/startGame/startGame.validations.ts [hunk 2]
```diff
@@ -1,23 +1,30 @@
...
+  ROWS_IS_INTEGER = 'rows must be a integer number',
```
Comment (danielglobant) [inline src/minesweeper/use-cases/startGame/startGame.validations.ts:9]: Is not integer, is a natural number. `Integer` includes negative numbers

File: src/minesweeper/use-cases/startGame/startGame.validations.ts [hunk 3]
```diff
@@ -1,23 +1,30 @@
...
+  COLUMNS_IS_INTEGER = 'columns must be a integer number',
```
Comment (danielglobant) [inline src/minesweeper/use-cases/startGame/startGame.validations.ts:10]: Is not integer, is a natural number. `Integer` includes negative numbers

File: src/minesweeper/infrastructure/controllers/startGame/startGame.controller.spec.ts [hunk 4]
```diff
@@ -1,16 +1,107 @@
...
+import { endpoints } from '@minesweeper/infrastructure/server/constants';
+import { GetQueryParamsErrorMessages } from '@minesweeper/infrastructure/server/utils/getQueryParams';
 
 describe('start game controller', () => {
-  it('should return correct data', async () => {
+  it('should response Bad Request Error if empty data is provided as query params', async () => {
     const server = createServer(requestHandler);
 
-    const response = await request(server).get('/start?bombs=1&rows=2&columns=2');
+    const response = await request(server).get(`${endpoints.game}`);
```
Comment (danielglobant) [inline src/minesweeper/infrastructure/controllers/startGame/startGame.controller.spec.ts:11]: you can type `response` and avoid this error

Comment (danielglobant) [reply src/minesweeper/infrastructure/controllers/startGame/startGame.controller.spec.ts:11]: https://www.npmjs.com/package/@types/supertest

File: src/minesweeper/use-cases/startGame/startGame.validations.ts [hunk 5]
```diff
@@ -1,23 +1,30 @@
...
 }
 
 export const makeValidations = (bombsInput: number, rows: number, columns: number): void => {
-  if (bombsInput < 1) throw new MinesweeperError(ErrorStartGame.BOMBS_GREATER_THAN_ZERO);
+  if (!Number.isInteger(bombsInput))
+    throw new MinesweeperError(StartGameErrorMessages.BOMBS_IS_INTEGER);
 
-  if (rows < 1) throw new MinesweeperError(ErrorStartGame.ROWS_GREATER_THAN_ZERO);
+  if (!Number.isInteger(rows)) throw new MinesweeperError(StartGameErrorMessages.ROWS_IS_INTEGER);
```
Comment (danielglobant) [inline src/minesweeper/use-cases/startGame/startGame.validations.ts:17]: integer to natural

File: src/minesweeper/use-cases/startGame/startGame.validations.ts (hunk 1 above)
Comment (Dylanflini) [reply src/minesweeper/use-cases/startGame/startGame.validations.ts:8]: yep, but the other validation only accepts values greater than 1

File: src/minesweeper/use-cases/startGame/startGame.validations.ts (hunk 2 above)
Comment (Dylanflini) [reply src/minesweeper/use-cases/startGame/startGame.validations.ts:9]: yep, but the other validation only accepts values greater than 1

File: src/minesweeper/use-cases/startGame/startGame.validations.ts (hunk 3 above)
Comment (Dylanflini) [reply src/minesweeper/use-cases/startGame/startGame.validations.ts:10]: yep, but the other validation only accepts values greater than 1

File: src/minesweeper/infrastructure/controllers/startGame/startGame.controller.spec.ts (hunk 4 above)
Comment (Dylanflini) [reply src/minesweeper/infrastructure/controllers/startGame/startGame.controller.spec.ts]: ready

**Review by danielglobant — APPROVED** [review]

**Pull request merged**